Feat: adiciona edição de sentimentos

ThaisMap opened the pull request:
Adiciona funcionalidade de edição de sentimentos com modal

Closes JOR-60

ThaisMap committed:
feat: adiciona modal para edição de hábitos

- substitui a página de edição e criação de hábitos por um modal

ThaisMap committed:
refactor: ajusta props do habitoCheckboxGroup

- tira a data das props do habitoCheckboxGroup
- tira o capitalize do título do modal de edição
- o eslint estava dando um aviso que o 'always-multiline' não era um valor válido

ThaisMap committed:
feat: adiciona funcionalidade de edição de sentimentos

kellynvd reviewed (changes requested):
@ThaisMap os mesmos problemas que comentei aqui https://github.com/grupotesseract/jornadasolar/pull/119 ocorrem para os sentimentos também, e além disso:

- [x] Não estou conseguindo adicionar um novo sentimento, ao clicar em concluir na modal de edição nada acontece e não aparece nenhuma mensagem de erro
- [x] Acredito que o botão "Criar novo" deveria aparecer antes do usuário clicar em Editar em cima da tela
- [x] Acho que poderia ter a animação de slide quando o usuário fecha a modal (ou clica em concluir também)
- [ ] O que acha de tentar extrair a lógica de controle de da modal de edição (e talvez da atualização) pra um outro componente pra não precisar repetir nos dois componentes de hábitos e sentimentos?

ThaisMap committed:
fix: ajusta validações

- valida se o emoji e o nome estão preenchidos
- valida se texto no input de emoji é um emoji mesmo
- valida pelo id se o hábito está marcado ao invés do nome

ThaisMap committed:
fix: desmarcar hábito editado e mensagem de erro

- Não permitia desmarcar hábito depois de editá-lo
- Não exibia mensagem de erro ao tentar criar hábito sem preencher os campos

ThaisMap committed:
Merge branch 'feat/modal-edicao-habitos' into feat/edicao-sentimentos

ThaisMap committed:
fix: ajustes na edição

- modal faz a animação de descer
- evita que o sentimento seja marcado ou desmarcado no background ao clicar pra editar
- não exibe botão de criar novo no cadastro
- na área logada sempre exibe botão de criar novo

dodocinho committed:
fix(sentimento): criar novo

Remove cortamento de manter o úlitmo emoji criado ao clicar novamente no botao de criar novo

dodocinho committed:
Delete settings.json

dodocinho committed:
Delete README.md
